From 76f1625bae95212ec6d4bc1bd1c8ff1232150c48 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 29 Aug 2022 13:14:09 -0700
Subject: [PATCH] Provide parameter prototype for functions without parameter

Fixes build with clang-15 with -Wstrict-prototypes

src/nautilus-toolbar.c:1205:22: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
nautilus_toolbar_new ()
                     ^
                      void

Upstream-Status: Backport [https://gitlab.gnome.org/GNOME/nautilus/-/commit/95d35aedf68f0398a6ee9e0b0af9ce80528fdc22]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/nautilus-file-undo-manager.c | 4 ++--
 src/nautilus-list-model.c        | 2 +-
 src/nautilus-starred-directory.c | 2 +-
 src/nautilus-toolbar.c           | 2 +-
 src/nautilus-view-model.c        | 2 +-
 5 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/nautilus-file-undo-manager.c b/src/nautilus-file-undo-manager.c
index 2a886176f..f2d1eebcf 100644
--- a/src/nautilus-file-undo-manager.c
+++ b/src/nautilus-file-undo-manager.c
@@ -258,13 +258,13 @@ nautilus_file_undo_manager_get_state (void)
 
 
 gboolean
-nautilus_file_undo_manager_is_operating ()
+nautilus_file_undo_manager_is_operating (void)
 {
     return undo_singleton->is_operating;
 }
 
 NautilusFileUndoManager *
-nautilus_file_undo_manager_get ()
+nautilus_file_undo_manager_get (void)
 {
     return undo_singleton;
 }
diff --git a/src/nautilus-list-model.c b/src/nautilus-list-model.c
index 7e2aeeeee..741007105 100644
--- a/src/nautilus-list-model.c
+++ b/src/nautilus-list-model.c
@@ -1625,7 +1625,7 @@ nautilus_list_model_get_drag_view (NautilusListModel *model,
 }
 
 GtkTargetList *
-nautilus_list_model_get_drag_target_list ()
+nautilus_list_model_get_drag_target_list (void)
 {
     GtkTargetList *target_list;
 
diff --git a/src/nautilus-starred-directory.c b/src/nautilus-starred-directory.c
index fd41418a4..07ed62693 100644
--- a/src/nautilus-starred-directory.c
+++ b/src/nautilus-starred-directory.c
@@ -556,7 +556,7 @@ nautilus_starred_directory_class_init (NautilusFavoriteDirectoryClass *klass)
 }
 
 NautilusFavoriteDirectory *
-nautilus_starred_directory_new ()
+nautilus_starred_directory_new (void)
 {
     NautilusFavoriteDirectory *self;
 
diff --git a/src/nautilus-toolbar.c b/src/nautilus-toolbar.c
index 5fe4e63a4..15a696827 100644
--- a/src/nautilus-toolbar.c
+++ b/src/nautilus-toolbar.c
@@ -1202,7 +1202,7 @@ nautilus_toolbar_class_init (NautilusToolbarClass *klass)
 }
 
 GtkWidget *
-nautilus_toolbar_new ()
+nautilus_toolbar_new (void)
 {
     return g_object_new (NAUTILUS_TYPE_TOOLBAR,
                          NULL);
diff --git a/src/nautilus-view-model.c b/src/nautilus-view-model.c
index 84c40302d..2935809ae 100644
--- a/src/nautilus-view-model.c
+++ b/src/nautilus-view-model.c
@@ -190,7 +190,7 @@ compare_data_func (gconstpointer a,
 }
 
 NautilusViewModel *
-nautilus_view_model_new ()
+nautilus_view_model_new (void)
 {
     return g_object_new (NAUTILUS_TYPE_VIEW_MODEL, NULL);
 }
